internal/jobs/common.go

```diff
@@ -20,10 +20,22 @@ func decodeJob(ctx context.Context, job dejq.Job, argInterface interface{}) erro
 	return nil
 }
 
-func contextLogger(ctx context.Context, jobName string, args interface{}, accountId, reservationId int64) context.Context {
+func contextLogger(ctx context.Context, accountId, reservationId int64) context.Context {
+	logger := ctxval.Logger(ctx).With().
+		Int64("reservation_id", reservationId).
+		Int64("account_id", accountId).Logger()
+
+	ad := dao.GetAccountDao(ctx)
+	account, err := ad.GetById(ctx, accountId)
+	if err != nil {
+		logger.Warn().Msgf("Unable to fetch account info for: %d", accountId)
+	} else {
+		logger = logger.With().
+			Str("account_number", account.AccountNumber.String).
+			Str("org_id", account.OrgID).Logger()
+	}
+
 	newContext := ctxval.WithAccountId(ctx, accountId)
-	logger := ctxval.Logger(newContext).With().Int64("reservation_id", reservationId).Logger()
-	logger.Info().Interface("args", args).Msgf("Processing job: '%s'", jobName)
```

Comment:
I quite liked this line, it was obvious what job has started with what args, why are we removing it? :)

Reply:
Because this is actually done here:

```go
func contextLogger(ctx context.Context, accountId, reservationId int64) context.Context {
	logger := ctxval.Logger(ctx).With().
		Int64("reservation_id", reservationId).
		Int64("account_id", accountId).Logger()

	ad := dao.GetAccountDao(ctx)
	account, err := ad.GetById(ctx, accountId)
	if err != nil {
		logger.Warn().Msgf("Unable to fetch account info for: %d", accountId)
	} else {
		logger = logger.With().
			Str("account_number", account.AccountNumber.String).
			Str("org_id", account.OrgID).Logger()
	}

	newContext := ctxval.WithAccountId(ctx, accountId)
	newContext = ctxval.WithLogger(newContext, &logger)
	return newContext
}
```

Every job log should now have all the fields as I describe in the PR description automatically, actually I see few more of these `Int64("reservation_id"` calls which I will remove now.

Reply:
I'm talking about the first log line at the start of every job, that is now removed, isn't it?

Reply:
Perhaps, I am not entirely sure which one you need, my intentions were not to remove anything useful, but I moved some logging messages from service so common functions where they do have more information available, e.g. job_id. Example from api process:

```
10:27AM DBG Started POST request /api/provisioning/v1/reservations/noop binary=pbapi bytes_in=0 hostname=mone.home.lan method=POST remote_ip=127.0.0.1:49619 trace_id=77a9fa786afacb1e602324182fe4e5fa url=/api/provisioning/v1/reservations/noop version=HEAD
10:27AM TRC Account found via org id: 13446659 binary=pbapi bytes_in=0 hostname=mone.home.lan method=POST remote_ip=127.0.0.1:49619 trace_id=77a9fa786afacb1e602324182fe4e5fa url=/api/provisioning/v1/reservations/noop version=HEAD
10:27AM TRC Account cache hit account=3 binary=pbapi bytes_in=0 hostname=mone.home.lan method=POST remote_ip=127.0.0.1:49619 trace_id=77a9fa786afacb1e602324182fe4e5fa url=/api/provisioning/v1/reservations/noop version=HEAD
10:27AM DBG Created a new reservation 30 account_id=3 account_number=6395343 binary=pbapi bytes_in=0 hostname=mone.home.lan method=POST org_id=13446659 remote_ip=127.0.0.1:49619 trace_id=77a9fa786afacb1e602324182fe4e5fa url=/api/provisioning/v1/reservations/noop version=HEAD
10:27AM INF Enqueuing job type: no_operation account_id=3 account_number=6395343 binary=pbapi bytes_in=0 hostname=mone.home.lan job_args={"AccountID":3,"Fail":false,"ReservationID":30} job_id=2003d977-6cf4-406e-a305-54ba3bb2ef0b job_type=no_operation method=POST org_id=13446659 remote_ip=127.0.0.1:49619 trace_id=77a9fa786afacb1e602324182fe4e5fa url=/api/provisioning/v1/reservations/noop version=HEAD
10:27AM INF Pushed job successfully account_id=3 account_number=6395343 binary=pbapi bytes_in=0 hostname=mone.home.lan job_args={"AccountID":3,"Fail":false,"ReservationID":30} job_id=2003d977-6cf4-406e-a305-54ba3bb2ef0b job_result=1 job_type=no_operation method=POST org_id=13446659 remote_ip=127.0.0.1:49619 trace_id=77a9fa786afacb1e602324182fe4e5fa url=/api/provisioning/v1/reservations/noop version=HEAD
10:27AM INF Completed POST request /api/provisioning/v1/reservations/noop in 19ms with 200 binary=pbapi bytes_in=0 bytes_out=22 hostname=mone.home.lan latency_ms=18.738125 method=POST remote_ip=127.0.0.1:49619 status=200 trace_id=77a9fa786afacb1e602324182fe4e5fa url=/api/provisioning/v1/reservations/noop version=HEAD
```

And worker:

```
12:00PM INF Dequeued job from Redis binary=pbworker hostname=mone.home.lan job_args={"AccountID":3,"Fail":false,"ReservationID":31} job_id=7f460d23-9ff0-4471-9ac2-8a576bf21aba job_type=no_operation version=HEAD
12:00PM DBG Reservation status change: 'No operation started' account_id=3 account_number=6395343 binary=pbworker hostname=mone.home.lan job_id=7f460d23-9ff0-4471-9ac2-8a576bf21aba org_id=13446659 reservation_id=31 step=true version=HEAD
12:00PM INF No operation finished account_id=3 account_number=6395343 binary=pbworker hostname=mone.home.lan job_id=7f460d23-9ff0-4471-9ac2-8a576bf21aba org_id=13446659 reservation_id=31 version=HEAD
12:00PM DBG Reservation status change: 'No operation finished' account_id=3 account_number=6395343 binary=pbworker hostname=mone.home.lan job_id=7f460d23-9ff0-4471-9ac2-8a576bf21aba org_id=13446659 reservation_id=31 step=true version=HEAD
12:00PM TRC Increased step number by: 1 account_id=3 account_number=6395343 binary=pbworker hostname=mone.home.lan job_id=7f460d23-9ff0-4471-9ac2-8a576bf21aba org_id=13446659 reservation_id=31 step=true version=HEAD
12:00PM DBG Job step: 1/1 account_id=3 account_number=6395343 binary=pbworker hostname=mone.home.lan job_id=7f460d23-9ff0-4471-9ac2-8a576bf21aba org_id=13446659 reservation_id=31 version=HEAD
12:00PM INF All jobs executed, marking job as success account_id=3 account_number=6395343 binary=pbworker hostname=mone.home.lan job_id=7f460d23-9ff0-4471-9ac2-8a576bf21aba org_id=13446659 reservation_id=31 version=HEAD
```

I feel like logging is now pretty nice, if you miss anything, let me know.